### PR TITLE
Attributes: exclusively lowercase A-Z in attribute names

### PR DESCRIPTION
**Pull request by timmywil**
Fixes gh-2730

### CODE REVIEW

File: test/unit/attributes.js [hunk 1]
```diff
@@ -451,7 +451,7 @@ QUnit.test( "attr(String, Object)", function( assert ) {
 
 	$radio = jQuery( "<input>", {
 		"value": "sup",
-		"type": "radio"
+		"TYPE": "radio"
```
Comment (mzgol) [inline test/unit/attributes.js]: What's the reason for this change?

Comment (timmywil) [inline test/unit/attributes.js]: To make sure we use the type attrhook even if the name that comes in is not lowercase.

Comment (mzgol) [inline test/unit/attributes.js]: Do we have one that tests the lowercase version, just in case?

File: src/attributes/attr.js [hunk 2]
```diff
@@ -7,7 +7,14 @@ define( [
 ], function( jQuery, access, support, rnotwhite ) {
 
 var boolHook,
-	attrHandle = jQuery.expr.attrHandle;
+	attrHandle = jQuery.expr.attrHandle,
+
+	// Exclusively lowercase A-Z in attribute names (gh-2730)
+	// https://dom.spec.whatwg.org/#converted-to-ascii-lowercase
+	raz = /[A-Z]/g,
```
Comment (dmethvin) [inline src/attributes/attr.js]: Not that it makes a lot of difference, but you could use `[A-Z]+` here to reduce the potential number of function calls?

File: test/unit/attributes.js (hunk 1 above)
Comment (dmethvin) [inline test/unit/attributes.js]: Do we need that since the `value` case above tests all-lowercase? Or is there some specific concern about `type`?

Comment (timmywil) [inline test/unit/attributes.js]: Yea, lowercase is covered elsewhere. I wanted to have one place where we ensure we still use the `type` attrHook even when the name passed is not lowercase. I'll add a comment.

File: src/attributes/attr.js (hunk 2 above)
Comment (timmywil) [inline src/attributes/attr.js]: Yup

**Comment by dmethvin** [comment]
LGTM
